pkg/operator/ceph/csi/cluster_config.go

```diff
@@ -155,24 +174,24 @@ func CreateCsiConfigMap(namespace string, clientset kubernetes.Interface, ownerI
 // value that is provided to ceph-csi uses in the storage class.
 // The locker l is typically a mutex and is used to prevent the config
 // map from being updated for multiple clusters simultaneously.
-func SaveClusterConfig(
-	clientset kubernetes.Interface, clusterNamespace string,
-	clusterInfo *cephclient.ClusterInfo, l sync.Locker) error {
+func SaveClusterConfig(clientset kubernetes.Interface, clusterNamespace string, clusterInfo *cephclient.ClusterInfo, l sync.Locker, newCsiClusterConfigEntry CsiClusterConfigEntry) error {
```

Comment:
How about we use a package var to lock? If callers don't use the same lock, different controllers could attempt to update the configmap at the same time, right?
```suggestion
func SaveClusterConfig(clientset kubernetes.Interface, clusterNamespace string, clusterInfo *cephclient.ClusterInfo, newCsiClusterConfigEntry CsiClusterConfigEntry) error {
```

Reply:
I see that callers are passing `&csi.ConfigMutex`. Why not make the mutex a private package var and no need to pass it as a param? This method could just use the package var directly, right?